plugins/module_utils/foreman_helper.py

```diff
@@ -771,6 +771,11 @@ def lookup_entity(self, key, params=None):
             # Already looked up or not an entity(_list) so nothing to do
             return self.foreman_params[key]
 
+        result = self._lookup_entity(self.foreman_params[key], entity_spec, params)
+        self.set_entity(key, result)
+        return result
+
+    def _lookup_entity(self, param, entity_spec, params=None):
```

Comment:
I know naming is hard, but having `param` and `params` in the same parameter list doesn't look sane.

How about `identifier`?